junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/Node.java

```diff
@@ -38,6 +38,16 @@ default C prepare(C context) throws Exception {
 		return context;
 	}
 
+	/**
+	 * Cleanup the supplied {@code context} after to execution.
+	 *
+	 * <p>The default implementation invokes {@link EngineExecutionContext#close()}
+	 * on the supplied context.
+	 */
+	default void cleanup(C context) {
```

Comment:
I think we should add `throws Exception` to the signature to be consistent with the other methods.

Reply:
Ok.

Reply:
Not ok. `javac` warns now:

```
EngineExecutionContext.java:25: warning: [try] auto-closeable resource EngineExecutionContext has a member method close() that could throw InterruptedException
public interface EngineExecutionContext extends AutoCloseable {
```

Reply:
Removed `extends AutoCloseable` and all is fine.

Reply:
My last two comment belong to `EngineExecutionContext`.